tensilelite/rocisa/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# ########################################################################
+# Copyright (C) 2025 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# ########################################################################
+cmake_minimum_required(VERSION 3.15)
+project(rocisa LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
+if(DEFINED Python_EXECUTABLE AND Python_EXECUTABLE)
+    message(STATUS "Manually set Python_EXECUTABLE to ${Python_EXECUTABLE}")
+else()
+    set(Python_EXECUTABLE python3)
+endif()
+
+# find nanobind
+execute_process(
+    COMMAND ${Python_EXECUTABLE} -m nanobind --cmake_dir
+    OUTPUT_VARIABLE nanobind_DIR
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_VARIABLE error
+    RESULT_VARIABLE result
+)
+if(result)
+    message(FATAL_ERROR "Did you forget to install nanobind?")
+endif()
+message(STATUS "Nanobind path: ${nanobind_DIR}")
+
+
+
+# Use cmake -Dpybind11_DIR=$(python3 -m nanobind --cmakedir) .. to set pybind dir
+# Use cmake -DCMAKE_PREFIX_PATH=$(python3 -m nanobind --cmake_dir) .. to set nanobind
+
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
```

Comment:
You want Development.Module, not Development. There are many Python installations that are not set up for `Development` (which implies the ability to embed a Python interpreter and other things), whereas `Development.Module` supports only what is needed to compile extensions (which is pretty much always enabled).

Also, you really want to move this to the top before your first use of `Python_EXECUTABLE` (this is the thing which ensures that is correctly populated, and you are using it above).

Reply:
the find_package has to be written in this exact way or else nanobind will return an error.

Reply:
I'm pretty sure that is not the root cause. But ok. This will fail as written for some variations of python, but I guess can figure that out later. This stuff is fiddly.

Reply:
Seems like it also works with ``Development.Module``. Thanks.